pkg/apis/allocation/v1/gameserverallocation.go

```diff
@@ -484,6 +491,53 @@ func (mp *MetaPatch) Validate() ([]metav1.StatusCause, bool) {
 	return causes, len(causes) == 0
 }
 
+// Counter is a Priority SortType
+const Counter string = "counter"
+
+// List is a Priority SortType
+const List string = "list"
+
+// Ascending a Priority Order where the smaller count is preferred.
+const Ascending string = "ascending"
+
+// Descending a Priority Order where the larger count is preferred.
+const Descending string = "descending"
+
+// Priority is a sorting option for GameServers with Counters or Lists based on the count or
+// number of items in a List.
+// SortType: Sort by a "counter" or a "list".
+// Key: The name of the Counter or List. If not found on the GameServer, has no impact.
+// Order: Sort by "ascending" or "descending". Default is "descending" so bigger count is preferred.
+// "ascending" would be smaller count is preferred.
+type Priority struct {
+	SortType string `json:"sortType"`
+	Key      string `json:"key"`
+	Order    string `json:"order"`
+}
+
+// Validate returns if the Priority is valid.
+func (p *Priority) validate(field string) ([]metav1.StatusCause, bool) {
+	var causes []metav1.StatusCause
+
+	if !(p.SortType == Counter || p.SortType == List) {
+		causes = append(causes, metav1.StatusCause{
+			Type:    metav1.CauseTypeFieldValueInvalid,
+			Message: "Priority.Sort must be either `counter` or `list`",
+			Field:   field,
+		})
+	}
+
+	if !(p.Order == Ascending || p.Order == Descending || p.Order == "") {
```

Comment:
I'm assuming Order will be defaulted, so checking for "" is probably not something we need to do? (or want to if we have defaulting)

Reply:
I haven't done any defaulting. Where in the code should I do defaulting for this?